src/common.speech/SynthesisTurn.ts

```diff
@@ -239,15 +258,27 @@ export class SynthesisTurn {
         Events.instance.onEvent(event);
     }
 
-    private updateTextOffset(text: string): void {
-        if (this.privTextOffset >= 0) {
+    private updateTextOffset(text: string, type: MetadataType): void {
+        if (type === MetadataType.WordBoundary) {
             this.privTextOffset = this.privRawText.indexOf(text, this.privNextSearchTextIndex);
             if (this.privTextOffset >= 0) {
                 this.privNextSearchTextIndex = this.privTextOffset + text.length;
+                if (this.privIsSSML) {
+                    if (this.privRawText.indexOf("<", this.privTextOffset + 1) > this.privRawText.indexOf(">", this.privTextOffset + 1)
+                        && !(text.length >= 2 && text[0] === "<" && text[text.length - 1] === ">")) {
```

Comment:
Cleaner to extract the second part of this check to a const at the beginning of the function since it's not expensive even if not needed, something like:
const isNotBracket: boolean = !(text.length >= 2 && text[0] === "<" && text[text.length - 1] === ">");

And you can reuse isNotBracket (or whatever name you choose) on line 279 below.

Reply:
thx, I moved the two check to functions.